pkg/resource/defaultclass.go

```diff
@@ -96,36 +95,50 @@ func (r *DefaultClassReconciler) Reconcile(req reconcile.Request) (reconcile.Res
 		return reconcile.Result{}, errors.Wrap(IgnoreNotFound(err), errGetClaim)
 	}
 
-	// Get resource classes with claim kind as default
-	classes := &corev1alpha1.ResourceClassList{}
-	if err := r.client.List(ctx, r.options, classes); err != nil {
-		// If this is the first time we encounter no defaults we'll be
+	// Get policies for claim kind in claim's namespace
+	policies := &unstructured.UnstructuredList{}
+	policies.SetGroupVersionKind(r.newPolicyList().GetObjectKind().GroupVersionKind())
+	options := &client.ListOptions{
+		Namespace: req.Namespace,
+	}
+	if err := r.client.List(ctx, options, policies); err != nil {
+		// If this is the first time we encounter listing error we'll be
 		// requeued implicitly due to the status update. If not, we don't
 		// care to requeue because list parameters will not change.
 		claim.SetConditions(corev1alpha1.ReconcileError(errors.New(errFailedList)))
 		return reconcile.Result{}, errors.Wrap(IgnoreNotFound(r.client.Status().Update(ctx, claim)), errUpdateClaimStatus)
 	}
 
 	// Check to see if no defaults defined for claim kind.
-	if len(classes.Items) == 0 {
-		// If this is the first time we encounter no defaults we'll be
+	if len(policies.Items) == 0 {
+		// If this is the first time we encounter no policies we'll be
 		// requeued implicitly due to the status update. If not, we will requeue
-		// after a time to see if a default class has been created.
-		claim.SetConditions(corev1alpha1.ReconcileError(errors.New(errNoDefaultClass)))
+		// after a time to see if apolicy has been created.
+		claim.SetConditions(corev1alpha1.ReconcileError(errors.New(errNoPolicies)))
 		return reconcile.Result{RequeueAfter: defaultClassWait}, errors.Wrap(IgnoreNotFound(r.client.Status().Update(ctx, claim)), errUpdateClaimStatus)
 	}
 
-	// Check to see if multiple defaults defined for claim kind.
-	if len(classes.Items) > 1 {
-		// If this is the first time we encounter multiple defaults we'll be
+	// Check to see if multiple policies defined for claim kind.
+	if len(policies.Items) > 1 {
+		// If this is the first time we encounter multiple policies we'll be
 		// requeued implicitly due to the status update. If not, we will requeue
-		// after a time to see if only one default class exists.
-		claim.SetConditions(corev1alpha1.ReconcileError(errors.New(errMultipleDefaultClasses)))
+		// after a time to see if only one policy class exists.
+		claim.SetConditions(corev1alpha1.ReconcileError(errors.New(errMultiplePolicies)))
 		return reconcile.Result{RequeueAfter: defaultClassWait}, errors.Wrap(IgnoreNotFound(r.client.Status().Update(ctx, claim)), errUpdateClaimStatus)
 	}
 
+	// Make sure single item is of correct policy kind
+	policy := r.newPolicy()
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(policies.Items[0].UnstructuredContent(), policy); err != nil {
```

Comment:
https://github.com/crossplaneio/crossplane/pull/607#discussion_r308924052

Capturing this comment from the previous PR here. Any thoughts?

Reply:
Was experiencing issues with `oc.Conversion` and found the error is returned difficult to interpret, so I ended up sticking with this pattern. Happy to revisit if you think it necessary!

Reply:
I think I would prefer if we used `runtime.Scheme` here.

Initially I preferred it only for aesthetics, which isn't enough reason to change this line especially given it uses the `DefaultUnstructuredConverter` under the hood. However it just occurred to me that if we pass a runtime.Scheme` as an `ObjectConverter` we can test this currently untested error condition by passing in a fake `ObjectConverter` that returns an error.